addons/mail/models/discuss/mail_thread.py

```diff
@@ -0,0 +1,32 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from markupsafe import Markup
+
+from odoo import models, _
+
+
+class MailThread(models.AbstractModel):
+    _inherit = "mail.thread"
+
+    def _notify_thread(self, message, msg_vals=False, **kwargs):
+        rdata = super()._notify_thread(message, msg_vals, **kwargs)
+        self._notify_mentioned_channel(**kwargs)
+        return rdata
+
+    def _notify_mentioned_channel(self, channel_ids=None, **kwargs):
+        """ Notify channels that have been mentioned in message"""
+        if channel_ids:
+            body = Markup(_("%(author)s mentioned this channel in %(record)s")) % {
+                "author": self.env.user.partner_id._get_html_link(),
+                "record": Markup('<a href="#" data-oe-type="notification" data-oe-id="%s" data-oe-model="%s">%s</a>') % (
+                    self.id, self._name, self.display_name
+                )
+            }
+            for channel in self.env['discuss.channel'].browse(channel_ids):
+                channel.message_post(body=body, message_type="notification", subtype_xmlid="mail.mt_comment")
```

Comment:
I would provide `message.id` rather than `thread.id`, so that you can call `/mail/message/<int:message_id>` which redirects to the message in its thread. That way the message that mentioned the channel is highlighted

Reply:
IMO we should use the `data-oe` format to open the message in the same location. Inserting a direct URL would redirect to a new route, where the breadcrumbs won't be visible. Using `data-oe` format helps to  maintain context

Reply:
Using route requires less LOCs and gives more guarantee that the feature works now and in the future. The aspect of opening another tab is more about the current decision with how all URL should be open in a different tab, which should has to be challenged in the future for a subset of URLs notably internal links.